query/src/main/scala/filodb/query/exec/BinaryJoinExec.scala

```diff
@@ -51,11 +51,11 @@ final case class BinaryJoinExec(queryContext: QueryContext,
   require(!on.contains(metricColumn), "On cannot contain metric name")
 
   val onLabels = on.map(Utf8Str(_)).toSet
+  val withExtraOnLabels = onLabels ++ Seq("_pi_", "_step_") // needed since these are internal tags
```

Comment:
Adding only new tag ‘pi’ is not sufficient? 

Also, do we have to keep this forever? Or for a short term would suffice?

Reply:
Technically yes. But this issue will also occur when customers change the publish interval themselves. Joins will fail and they have to provide this internal tag explicitly. Instead, we could do it automatically for them.